lib/src/main/kotlin/com/lemonappdev/konsist/core/declaration/KoComplexDeclarationImpl.kt

```diff
@@ -7,7 +7,8 @@ import org.jetbrains.kotlin.psi.KtClassOrObject
 
 internal abstract class KoComplexDeclarationImpl(
     private val ktClassOrObject: KtClassOrObject,
-) : KoDeclarationImpl(ktClassOrObject),
+    override val parent: KoBaseDeclarationImpl,
```

Comment:
`parent: KoBaseDeclaration`

and the `parent` property in `KoDeclarationImpl constructor` should not be opened